apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/SettingsActivity.java

```diff
@@ -311,6 +311,122 @@ public void setEnabled(boolean enabled) {
                     Distribute.setEnabledForDebuggableBuild(enabled);
                 }
             });
+            final IsEnabled updateTrackBeforeStartIsEnabled = new IsEnabled() {
+
+                @Override
+                public boolean isEnabled() {
+                    return MainActivity.sSharedPreferences.getBoolean(getString(R.string.appcenter_distribute_update_track_before_start_key), false);
+                }
+
+                @Override
+                public String getSummary() {
+                    return getString(this.isEnabled() ? R.string.appcenter_distribute_update_track_before_start_enabled : R.string.appcenter_distribute_update_track_before_start_disabled);
+                }
+            };
+            initChangeableSetting(R.string.appcenter_distribute_update_track_before_start_key, updateTrackBeforeStartIsEnabled.getSummary(), new Preference.OnPreferenceChangeListener() {
+
+                @Override
+                public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    if (newValue == null) {
+                        return false;
+                    }
+                    String[] whenToUpdateTrackEntries = getResources().getStringArray(R.array.appcenter_distribute_when_to_update_track_entries);
+                    boolean updateTrackBeforeStartNewValue = newValue.toString().equals(whenToUpdateTrackEntries[1]);
+                    MainActivity.sSharedPreferences.edit().putBoolean(getString(R.string.appcenter_distribute_update_track_before_start_key), updateTrackBeforeStartNewValue).apply();
+
+                    /*
+                     * TODO when updating the demo during release process:
+                     *  1) Replace the next line with 'int currentTrack = Distribute.getUpdateTrack();'
+                     *  2) remove try/catch block
+                     */
+                    int currentTrack = 1;
+                    try {
+                        Method getUpdateTrackMethod = Distribute.class.getMethod("getUpdateTrack");
+                        currentTrack = (int) getUpdateTrackMethod.invoke(null);
+                    } catch (Exception e) {
+                        Toast.makeText(getActivity(), "No Update Track API in this build", Toast.LENGTH_SHORT).show();
+                    }
+                    MainActivity.sSharedPreferences.edit().putInt(getString(R.string.appcenter_distribute_update_track_before_start_chosen_track), currentTrack).apply();
+                    preference.setSummary(updateTrackBeforeStartIsEnabled.getSummary());
+                    return true;
+                }
+            });
+            final IsEnabled updateTrackIsEnabled = new IsEnabled() {
+
+                @Override
+                public boolean isEnabled() {
```

Comment:
since this interface is made up in this PR, the boolean and enabled semantics are wrong here. It's an update track enum, not bool isEnabled.